cmd/anonymizer/app/uiconv/extractor.go

```diff
@@ -63,7 +64,7 @@ func (e *Extractor) Run() error {
 			spans = append(spans, *span)
 		}
 	}
-	if err != io.EOF {
+	if !errors.Is(err, io.EOF) {
 		return fmt.Errorf("failed when scanning the file: %w", err)
 	}
 	trace := uimodel.Trace{
```

Comment:
hm, a test in this pkg ran into NPE (I don't recall it being flaky)
```
--- FAIL: TestExtractor_TraceScanError (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x9e87fd]

goroutine 23 [running]:
testing.tRunner.func1.2({0xa387c0, 0xe3df40})
	/opt/hostedtoolcache/go/1.21.5/x64/src/testing/testing.go:1545 +0x3f7
testing.tRunner.func1()
	/opt/hostedtoolcache/go/1.21.5/x64/src/testing/testing.go:1548 +0x716
panic({0xa387c0?, 0xe3df40?})
	/opt/hostedtoolcache/go/1.21.5/x64/src/runtime/panic.go:920 +0x270
github.com/jaegertracing/jaeger/cmd/anonymizer/app/uiconv.(*Extractor).Run(0xc0001afb00)
	/home/runner/work/jaeger/jaeger/cmd/anonymizer/app/uiconv/extractor.go:80 +0x73d
github.com/jaegertracing/jaeger/cmd/anonymizer/app/uiconv.TestExtractor_TraceScanError(0x0?)
	/home/runner/work/jaeger/jaeger/cmd/anonymizer/app/uiconv/extractor_test.go:108 +0x57f
testing.tRunner(0xc0001b31e0, 0xad75e8)
	/opt/hostedtoolcache/go/1.21.5/x64/src/testing/testing.go:1595 +0x262
created by testing.(*T).Run in goroutine 1
	/opt/hostedtoolcache/go/1.21.5/x64/src/testing/testing.go:1648 +0x846
```

Reply:
debugging:
```
=== RUN   TestExtractor_TraceScanError
err != io.EOF true
!errors.Is(err, io.EOF) false
```

Reply:
fixed in 249c9592 by using a custom error instead of `io.EOF` which was clashing when using `error.Is`